src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -109,11 +109,11 @@ PipeReader IRequestBodyPipeFeature.Reader
                 if (!ReferenceEquals(_requestStreamInternal, RequestBody))
                 {
                     _requestStreamInternal = RequestBody;
-                    RequestBodyPipeReader = PipeReader.Create(RequestBody);
+                    RequestBodyPipeReader = PipeReader.Create(RequestBody, new StreamPipeReaderOptions(_context.MemoryPool, _context.MemoryPool.GetMinimumSegmentSize(), _context.MemoryPool.GetMinimumAllocSize()));
 
                     OnCompleted((self) =>
                     {
-                        ((PipeWriter)self).Complete();
+                        ((PipeReader)self).Complete();
```

Comment:
ouch

Reply:
Why isn't this defined like `OnCompleted<T>(Func<T, Task>, T)`?

Reply:
We messed up

Reply:
I guess that's a good reason